x/merkledb/trie_test.go

```diff
@@ -1178,3 +1180,179 @@ func TestTrieViewInvalidChildrenExcept(t *testing.T) {
 	require.True(view3.invalidated)
 	require.Empty(view1.childViews)
 }
+
+func Test_Trie_ConcurrentReadWrite(t *testing.T) {
+	require := require.New(t)
+
+	trie, err := getBasicDB()
+	require.NoError(err)
+	require.NotNil(trie)
+	newTrie, err := trie.NewView()
+	require.NoError(err)
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := newTrie.Insert(context.Background(), []byte("key"), []byte("value"))
+		require.NoError(err)
+	}()
+
+	require.Eventually(
+		func() bool {
+			value, err := newTrie.GetValue(context.Background(), []byte("key"))
+
+			if err == database.ErrNotFound {
+				return false
+			}
+
+			require.NoError(err)
+			require.Equal([]byte("value"), value)
+			return true
+		},
+		time.Second,
+		time.Millisecond,
+	)
+}
+
+func Test_Trie_ConcurrentNewViewAndCommit(t *testing.T) {
+	require := require.New(t)
+
+	trie, err := getBasicDB()
+	require.NoError(err)
+	require.NotNil(trie)
+
+	newTrie, err := trie.NewView()
+	require.NoError(err)
+	err = newTrie.Insert(context.Background(), []byte("key"), []byte("value0"))
+	require.NoError(err)
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := newTrie.CommitToDB(context.Background())
+		require.NoError(err)
+	}()
+
+	newView, err := newTrie.NewView()
+	require.NoError(err)
+	require.NotNil(newView)
+}
+
+func Test_Trie_ConcurrentDeleteAndMerkleRoot(t *testing.T) {
+	require := require.New(t)
+
+	trie, err := getBasicDB()
+	require.NoError(err)
+	require.NotNil(trie)
+
+	newTrie, err := trie.NewView()
+	require.NoError(err)
+	err = newTrie.Insert(context.Background(), []byte("key"), []byte("value0"))
+	require.NoError(err)
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := newTrie.Remove(context.Background(), []byte("key"))
+		require.NoError(err)
+	}()
+
+	rootID, err := newTrie.GetMerkleRoot(context.Background())
+	require.NoError(err)
+	require.NotNil(rootID)
```

Comment:
Small question - because `rootID` is a `[32]byte`, I don't think it can ever be `nil`... Should we be verifying anything about what `rootID` is?